### ISSUE TITLE
TYPEORM_URL not working with cli

### ISSUE THREAD

**Issue by rynop** [typeorm/typeorm#2738]
**Issue type:**

[x] bug report

**Database system/driver:**

[x] `mysql` v5.6

**TypeORM version:**

[x] `latest`
 
I'm trying to run `typeorm schema:sync` with the following `.env` file but I get `Error: No connection options were found in any of configurations file`

```
TYPEORM_URL=mysql://root:root@127.0.0.1:3306/mydb
TYPEORM_LOGGING = true
TYPEORM_ENTITIES = /**/*.entity{.ts,.js}
```

If my `.env` has the following contents, it works:

```
TYPEORM_CONNECTION = mysql
TYPEORM_HOST = 127.0.0.1
TYPEORM_USERNAME = root
TYPEORM_PASSWORD = root
TYPEORM_DATABASE = mydb
TYPEORM_PORT = 3306
TYPEORM_LOGGING = true 
TYPEORM_ENTITIES = /**/*.entity{.ts,.js}
```

Is this a bug? I'm OK with having to be explicit when running the CLI - just want to make sure I can use the npm mysql `url` option like so (FWIW using nestjs):

```
TypeOrmModule.forRoot({
      type: 'mysql',
      url: process.env[MYSQL_URL],
      entities: [__dirname + '/**/*.entity{.ts,.js}'],
      synchronize: false,
    })
```

**Comment by jussikinnula** [comment]
I fixed the issue, there's PR waiting at: https://github.com/typeorm/typeorm/pull/2843

**Comment by jussikinnula** [comment]
Also, I did create a separate PR https://github.com/typeorm/typeorm/pull/2843 for getting connection type automatically from the same TYPEORM_URL (e.g. `postgres://hostname:5432/database` -> `postgres`).

**Comment by osdiab** [comment]
I am experiencing this and would really like to use this.

**Comment by osdiab** [comment]
Huh, looks like it's already been merged in - im on `0.2.11` and trying to run the `typeorm migrations:generate` command, with `TYPEORM_URL` it doesnt work but with the manual options it does seem to work (except for the `ssl` field missing, which i've addressed in #3409 )

**Comment by pleerock** [comment]
@jussikinnula can you please help?

**Comment by jussikinnula** [comment]
If I understood correctly, when using URL - the SSL option from URL parameters should be parsed also. I can fix & make PR.

The workaround was to use `TYPEORM_DRIVER_EXTRA='{ "ssl": true }'`.

**Comment by pleerock** [comment]
@jussikinnula no, ssl is parsed correctly, I think url isn't working for @osdiab can you please provide more details?

**Comment by osdiab** [comment]
Yes, exactly - if .env has the TYPEORM_URL field but none of the manual ones like TYPEORM_HOST, TYPEORM_PORT, etc then when I try to run migrations (that’s all I’m using the cli for right now) then it says that the connection is refused for 127.0.0.1:5432, even though TYPEORM_URL has a remote db defined.

But if I just split the URL into its individual pieces, then it connects just fine.

**Comment by jussikinnula** [comment]
Re @osdiab, sorry it's my fault I didn't test the cli when making the last change. So I missed the part that actually if `TYPEORM_CONNECTION` is not set, then the cli cannot use other env parameters.

The fix is here: https://github.com/typeorm/typeorm/pull/3421

And as a workaround, you only need to set `TYPEORM_CONNECTION` besides `TYPEORM_URL`. After the fix is applied, only `TYPEORM_URL` will be needed.

P.S. Related to the "SSL" thingie, I meant that on many Heroku addon databases they use `ssl=true` in the GET parameters in the URL. I guess those could be also supported by `typeorm`, at least they are not supported now. This would apply at least MongoDB.
